### PR TITLE
[PREVIEW] SL-1535 Tidy up creation of listing request

### PR DESCRIPTION
**Pull request by zielinskikamil**
Hey, in events I've introduced new models for creating, updating and returning hearing part.
The same approach I've applied on frontend side. 3 new models has been added to reflect required changes. This required to update HearingParty and HearingPartViewModel as well. 
This changed tided up a bit ListingCreate. I have removed duplicated userTransactionId

Also with these changes, I normalized out properties name from
session -> sessionId
caseType -> caseTypeCode
hearingType -> hearingTypeCode

these changed requires following PRs
https://github.com/hmcts/snl-events/pull/130
https://github.com/hmcts/snl-rules/pull/69

### CODE REVIEW

File: src/app/hearing-part/components/listing-create/listing-create.component.spec.ts [hunk 1]
```diff
@@ -49,10 +49,8 @@ let storeSpy: jasmine.Spy;
 let component: ListingCreateComponent;
 let store: Store<State>;
 let fixture: ComponentFixture<ListingCreateComponent>;
-let listingCreateNoteConfig: ListingCreateNotesConfiguration;
```
Comment (leszekgu) [inline src/app/hearing-part/components/listing-create/listing-create.component.spec.ts]: Why are you removing this and other notes related elements?

Comment (zielinskikamil) [reply src/app/hearing-part/components/listing-create/listing-create.component.spec.ts]: notes were passed to listing request effects and from there haven't used anymore. I just removed unnecessary code.

File: src/app/sessions/containers/sessions-listings-search/sessions-listings-search.component.spec.ts [hunk 2]
```diff
@@ -67,16 +69,33 @@ const mockedNotes: Note[] = [
...
+    sessionId: undefined,
     caseNumber: 'abc123',
     caseTitle: 'some-case-title',
     caseType: stubCaseType,
     hearingType: stubHearingType,
-    duration: moment.duration(sessionDuration),
-    scheduleStart: nowMoment,
-    scheduleEnd: nowMoment,
+    duration: moment.duration(durationStringFormat),
+    scheduleStart: moment(nowISOSting),
```
Comment (arompa) [inline src/app/sessions/containers/sessions-listings-search/sessions-listings-search.component.spec.ts]: Seems there's a typo - shouldn't it be String as in nowISOString?

Comment (zielinskikamil) [reply src/app/sessions/containers/sessions-listings-search/sessions-listings-search.component.spec.ts]: It is fine ;-)
duration is a time that tells how long it last, its also encoded is specific format eg  'PT30m'
scheduleStart and scheduleEnd makes a border when given hear part can occur

Comment (arompa) [reply src/app/sessions/containers/sessions-listings-search/sessions-listings-search.component.spec.ts]: I didn't mean changing the type. All I meant that there seems to be a typo - a spelling mistake ;)

**Pull request merged**